src/expr/src/scalar/func.rs

```diff
@@ -3521,20 +3509,97 @@ impl fmt::Display for BinaryFunc {
     }
 }
 
+/// A description of an SQL unary function that has the ability to lazy evaluate its arguments
```

Comment:
I realize this was something we have largely left unmentioned and undocumented, and you're just inheriting the problem by trying to make things better–– but! would you mind throwing in at least a line explaining when/why you would lazily vs. eagerly evaluate functions?

Reply:
I will write a longer form comment later, but a short explanation is that functions that lazy evaluate their arguments have the opportunity to stop evaluating early. The best example is the AND function where if the first argument evaluates to false then we can skip evaluating the second one. 

I can't think of a good example with unary functions because they can't really skip the evaluation so I guess the LazyUnaryFunc doesnt make a ton of sense but it has to exist because of how we store the expressions.

But these kind of functions are very few, hence the Eager variant of the trait that directly provides evaluated values to the implementors